### ISSUE TITLE
Clicking on any message put ALL message of all folder as there are in same thread

### ISSUE THREAD

**Issue by Freshou** [nextcloud/mail#5847]
### Expected behavior
Only show up e-mail that are of the same thread

### Actual behavior
If i click on any e-mail from list, i got on the right pane, 14 000 e-mail show in thread ... It take over 2 minutes to show up ... 

### Mail app

1.11.2

IMAP

### Server configuration

Ubuntu 20.04

Apache

MariaDB

Php 9

NextCloud 22.2.3

#### Client configuration
**Browser:** Any ... Tried in Safari, Google Chrome and Firefox, same thing... 

**Operating system:** Tried on Mac and Window

https://user-images.githubusercontent.com/7290335/146035419-9d73a1da-7ecf-471f-934f-732923249f98.mp4

s

**Comment by ChristophWurst** [comment]
This should have been fixed by https://github.com/nextcloud/mail/pull/5837

**Comment by ChristophWurst** [comment]
Could you run `occ maintenance:repair` and post the output?

**Comment by Freshou** [comment]
sudo -u www-data php8.0 occ maintenance:repair
Failed to load repair step for mail: Repair step 'OCA\Mail\Command\RepairMailTheads' is unknown
 - Repair MySQL collation
     - All tables already have the correct collation -> nothing to do
 - Repair mime types
 - Clean tags and favorites
     - 0 tags of deleted users have been removed.
     - 0 tags for delete files have been removed.
     - 0 tag entries for deleted tags have been removed.
     - 0 tags with no entries have been removed.
 - Repair invalid shares
 - Move .step file of updater to backup location
 - Add move avatar background job
     - Repair step already executed
 - Add preview cleanup background jobs
 - Fix potential broken mount points
     - No mounts updated
 - Repair language codes
 - Install new core bundle components
 - Add log rotate job
 - Clear frontend caches
     - Image cache cleared
     - SCSS cache cleared
     - JS cache cleared
 - Clear every generated avatar on major updates
 - Add preview background cleanup job
 - Queue a one-time job to cleanup old backups of the updater
 - Cleanup invalid photocache files for carddav
 - Add background job to cleanup login flow v2 tokens
 - Remove potentially over exposing share links
     - No need to remove link shares.
 - Clear access cache of projects
 - Reset generated avatar flag
 - Keep legacy encryption enabled
 - Check encryption key format
 - Remove old dashboard app config data
 - Add job to cleanup the bruteforce entries
 - Queue a one-time job to check for user uploaded certificates
 - Repair DAV shares
 - Add background job to set the lookup server share state for users
 - Add read permissions for existing announcements
    0 [->--------------------------]
 - Update name of the stored view
 - Upgrading Circles App
     - (Circles) Migrating to 22
    0 [--->------------------------]
    0 [----->----------------------]
 11/11 [============================] 100%

     - (Circles) Migrating to 22.1.x
    0 [-------------------->-------]
 - Fix component of birthday calendars
     - 6 birthday calendars updated.
 - Regenerating birthday calendars to use new icons and fix old birthday events without year
     - Adding background jobs to regenerate birthday calendar
 - Fix broken values of calendar objects
    0 [------------------------>---]
 - Registering building of calendar search index as background job
     - Add background job
 - Register building of social profile search index as background job
 - Registering background jobs to update cache for webcal calendars
     - Added 0 background jobs to update webcal calendars
 - Registering building of calendar reminder index as background job
     - Add background job
 - Clean up orphan event and contact data
     - 0 events without a calendar have been cleaned up
     - 0 properties without an events have been cleaned up
     - 0 changes without a calendar have been cleaned up
     - 0 cached events without a calendar subscription have been cleaned up
     - 0 changes without a calendar subscription have been cleaned up
     - 0 contacts without an addressbook have been cleaned up
     - 0 properties without a contact have been cleaned up
     - 0 changes without an addressbook have been cleaned up
 - Remove activity entries of private events
     - Removed 0 activity entries
 - Fix the share type of guest shares when migrating from ownCloud
 - Copy the share password into the dedicated column
 - Set existing shares as accepted
 - Copy Spacedeck package data
 - Restore default tags that are missing
     - Looking up default tags
     - 16 accounts to check found
 16/16 [============================] 100%
 - Add a generated message-id to all Mail messages that have none
     - Looking up messages without a message-id
 - Purify and migrate collected mail addresses
 123/123 [============================] 100%
 - Insert background jobs for all accounts
 16/16 [============================] 100%
 - Make Mail itinerary extractor executable
 - Create or update provisioned Mail accounts
     - 0 accounts provisioned
 - Update OAuth token expiration times
 - Polls - Create indices and foreign key constraints
     - Polls - Indices created.
     - Polls - Foreign key contraints created.
 - Create help command
 - Invalidate access cache for projects conversation provider
     - Invalidation not required
 - Cache the user display names
 - Switches from default updater server to the customer one if a valid subscription is available
 - Send an admin notification if monthly report is disabled
 - Add background job to check for backup codes
 - Populating added database structures for workflows

**Comment by ChristophWurst** [comment]
> Failed to load repair step for mail: Repair step 'OCA\Mail\Command\RepairMailTheads' is unknown

Bingo!

**Comment by Freshou** [comment]
Ok, ans ho do i correct this ?

**Comment by ChristophWurst** [comment]
We'll send a PR, backport the fix and push out a new release tomorrow morning.

If you are in a hurry you can use `UPDATE oc_mail_messages SET in_reply_to=NULL WHERE in_reply_to='<>'` on your db.

Make a backup just in case something goes wrong.

**Comment by Freshou** [comment]
The Quick Fix worked properly ! 

Waiting for tomorrow update ! 

Thanx !

**Comment by dcrobertson01** [comment]
Is there any way I can just see the email I clicked on? At the moment I am getting all mail from all mail boxes all the time. If the developers know what I want better than I do - eg not letting us disable  the 'priority' inboxes and not letting us select a default inbox - could they at least let us sort our mail in the order we want it sorted?

I keep thinking it would be too much of a hassle to change to round cube or squirrel mail - one of which I used to use before I changed to this mail app and it kept being 'improved' - but in the long run, maybe it would be easier.
